packages/material-ui/src/InputLabel/InputLabel.js

```diff
@@ -11,14 +11,17 @@ import { getInputLabelUtilityClasses } from './inputLabelClasses';
 
 const overridesResolver = (props, styles) => {
   const { styleProps } = props;
-  return deepmerge(styles.root, {
-    ...(!styleProps.formControl && styles.formControl),
-    ...(styleProps.size === 'small' && styles.sizeSmall),
-    ...(styleProps.shrink && styles.shrink),
-    ...(!styleProps.disableAnimation && styles.animated),
-    ...styles[styleProps.variant],
-    [`& .${formLabelClasses.asterisk}`]: styles.asterisk,
-  });
+  return deepmerge(
+    {
+      ...(!styleProps.formControl && styles.formControl),
+      ...(styleProps.size === 'small' && styles.sizeSmall),
+      ...(styleProps.shrink && styles.shrink),
+      ...(!styleProps.disableAnimation && styles.animated),
+      ...styles[styleProps.variant],
+      [`& .${formLabelClasses.asterisk}`]: styles.asterisk,
+    },
+    styles.root || {},
+  );
```

Comment:
Hi @oliviertassinari can you revisit this change again? This change causes a regression in our product.

After this change:

![image](https://user-images.githubusercontent.com/5390719/117270549-e922ef00-ae8b-11eb-830c-e9b49fa3a87b.png)

Revert this specific change in node_modules and it works well:

![image](https://user-images.githubusercontent.com/5390719/117270782-25564f80-ae8c-11eb-892e-3051bd5e5175.png)

I can't tell why this is happening but it seems very strange.

Reply:
@Jack-Works we changed how the `styleOverrides` work in the latest release alpha.32, can you try with this version? Also would be useful if you can share the code, preferably in an issue so that it would help other developers if they stumble upon it.

Reply:
Our product is open sourced but it's but hard to build and preview. If you'd like to have a try I can guide you how to setup the project. I'll try alpha 32 later, thanks!

Reply:
Let's try `alpha.32` and see, if not simple repro would be best for us to debug.

Reply:
Hmm, I found we're already on alpha 32... I can try to make a reproduce but I doubt if I can make a simple repro

Reply:
It appears to be a problem in our codebase. We overwrite MuiInputLabel in the theme. Remove them solves the problem

https://github.com/DimensionDev/Maskbook/pull/3095